otx/cli/manager/config_manager.py

```diff
@@ -243,6 +241,96 @@ def auto_task_detection(self, data_roots: str) -> str:
+            jpgs = glob.glob(f"{dir}/*.jpg")
+            return len(pngs) > 0 or len(jpgs) > 0
+
+        def _check_semisl_requirements(train_dir, unlabeled_dir, thershold=0.07):
+            """Check if quantity of unlabeled images is sufficient for Semi-SL learning."""
+            all_unlabeled_images = len(os.listdir(unlabeled_dir))
+            if all_unlabeled_images <= 1:
+                # the folder is empty or only 1 image there
+                return False
+            all_train_images = len(os.listdir(train_dir))
+            # check if number of unlabeled images is more than relative thershold
+            if all_unlabeled_images >= thershold * all_train_images:
+                return True
+            return False
+
+        # if user explicitly passed train type via args
+        if self.args.train_type is not None:
+            self.train_type = self.args.train_type
+            return
+
+        path_to_train_data = Path(self.args.train_data_roots)
+        if not Path.is_dir(path_to_train_data):
+            raise ValueError(
+                "train-data-roots isn't a directory or it doesn't exist. "
+                "Please, check command line and directory path."
+            )
+
+        if _check_is_only_images(path_to_train_data):
+            # If train folder with images only was passed to args
+            # Then we start self-supervised training
+            print("[*] Selfsupervised training type detected")
+            self.train_type = "Selfsupervised"
+            return
+
+        # if user explicitly passed unlabeled images folder
+        if self.args.unlabeled_data_roots is not None:
+            unlabeled_path = Path(self.args.unlabeled_data_roots)
+            if Path.is_dir(unlabeled_path):
+                if not _check_semisl_requirements(path_to_train_data, unlabeled_path):
+                    print(
+                        "WARNING: There are none or too litle images to start Semi-SL training."
+                        "It should be more that relative threshold"
+                        "Start Supervised training instead."
+                    )
+                self.train_type = "Semisupervised"
+                return
+            else:
+                raise ValueError(
+                    "Unlabeled images isn't a directory or it doesn't exist. "
+                    "Please, check command line and directory path."
+                )
+
+        unlabeled_folder_name = [
+            item
+            for item in os.listdir(path_to_train_data)
+            if (item.startswith("unlabeled") and os.path.isdir(os.path.join(path_to_train_data, item)))
+        ]
+        if unlabeled_folder_name:
+            # if in directory we have several unlabeled data folders -> retrive first
+            path_to_unlabeled_data = Path(path_to_train_data / unlabeled_folder_name[0])
+            if not _check_semisl_requirements(path_to_train_data, path_to_unlabeled_data):
+                print(
```

Comment:
As I understand it, this is a case where `unlabeled_data_roots` was **not taken** as an argument, but an unlabeled folder was found, and the image was determined to be insufficient.

Is a WARNING necessary in this case? I think it's more natural to print a WARNING when automatically declare `Semisupervised` outside of an if statement. (before line 328)

Isn't it more likely that users will think it's `Supervised` by default because user didn't declare `unlabeled_data_roots`?
I think the following is natural.
With unlabeled_data_roots -> Detect Incremental, Then WARNING
No unlabeled_data_roots -> Detect SemiSL, Then WARNING
(Of course, that could just be me.)

Reply:
I will add in the documentation (in the next PR, I think) expected behavior related to auto configuration. So, if user will have unlabeled data folder with images - it is expected, that SemiSL should be started. However, in some cases the number of images is insufficient to start SemiSL, so I added a warning

After this feature will be merged declaration of unlabeled_data_roots in arguments will be optional and for some users maybe it will be more convenient to create unlabeled_images in the data root and always omit unlabeled_data_roots declaration to start Semi-SL

However, I will add a message like: "The Semi-SL training type detected. {folder_name} will be used for unlabeled_data_roots"

Reply:
What I wanted to say, when _check_semisl_requirements return False - I think we should print a message with warning, that unlabeled images is not enough for good Semi-SL perfomance

Reply:
Since we have to omit this part of functionality - the conversation can be resolved